### PR TITLE
Fixed issue while publishing test reports if one of the file is invalid xml file

### PR DESCRIPTION
**Pull request by nigurr**
Issue: If one of the test report file is invalid, the logic to return the promise fails. Thus the job gets stuck.
Fix: Call all the report executions and then return the promise with final test run details

### CODE REVIEW

**Comment by nigurr** [comment]
UT is added and verified manually

File: src/agent/testrunpublisher.ts [hunk 1]
```diff
@@ -277,28 +277,29 @@ export class TestRunPublisher {
             defer.resolve(null);
         }
 
-        for (var i = 0; i < resultFiles.length; i++) {
-            var report = _this.readResults(resultFiles[i]).then(function(res) {
-                res.testResults.forEach(tr => {
-                    totalTestCaseDuration += +tr.durationInMs;
+        var promises = resultFiles.map(function(file) {
```
Comment (AshwiniChalla) [inline src/agent/testrunpublisher.ts]: check if resultFiles is not null above

**Comment by AshwiniChalla** [comment]
approved with suggestions

**Comment by acesiddhu** [comment]
go ahead

**Pull request merged**